File: EcoBookAiBackend/src/main/java/com/ecobook/dto/AuthResponseDTO.java
package com.ecobook.dto;

import com.fasterxml.jackson.databind.PropertyNamingStrategies;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Set;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
public class AuthResponseDTO {
    private String id;
    private String email;
    private String nome;
    private String whatsapp;
    private String cpf;
    private String cidade;
    private String bairro;
    private String instituicao;
    private String fotoPerfilUrl;
    private Boolean perfilCompleto;
    private Boolean consentimentoIa;
    private Set<String> necessidadesAcademicas;
    private String role;
    private String token;
    private Long expiresIn;
}

File: EcoBookAiAndroid/app/src/main/kotlin/com/ecobook/dto/AuthResponseDTO.kt
package com.ecobook.dto

data class AuthResponseDTO(
    val id: String,
    val email: String,
    val nome: String,
    val whatsapp: String? = null,
    val cidade: String? = null,
    val bairro: String? = null,
    val instituicao: String? = null,
    val perfilCompleto: Boolean,
    val consentimentoIa: Boolean = false,
    val role: String,
    val token: String,
    val expiresIn: Long
)
